tests/reproducibility/fixtures.py

```diff
@@ -39,3 +41,18 @@ def assert_reproducible(func, num_iter=20):
             model.get_weights(),
             model_new.get_weights(),
         )
+
+
+@contextmanager
+def use_seed(seed):
```

Comment:
I think typically a context manager in this style would try to be invisible: that is, `a` and `b` would be equal in

```python
a = random()
b = random()
```

```python
a = random()
with use_seed(0):
    something_complicated()
b = random()
```

which isn't the case as this is written. However, I think it _is_ possible with `get_state` and `set_state` APIs:

- https://docs.python.org/3/library/random.html#random.getstate
- https://docs.scipy.org/doc/numpy-1.17.0/reference/random/legacy.html#seeding-and-state

but there doesn't seem to be something of this sort for TF (https://www.tensorflow.org/api_docs/python/tf/random), and this is probably unavoidable, if the randomness fits into its graph.

Do you think this is an important property?

If not, I'm not sure this function is that valuable: to me, there doesn't seem to be much difference between an explicit `set_seed(None)`, and just continuing after a single `set_seed(seed)`. That is, instead of `with use_seed(0): ...` we could do `set_seed(0); ...` and have very similar behaviour. What do you think?

Reply:
> to me, there doesn't seem to be much difference between an explicit set_seed(None), and just continuing after a single set_seed(seed). That is, instead of with use_seed(0): ... we could do set_seed(0); ... and have very similar behaviour.

Hmm that's true 🤔  Okay I think I'll just remove this thing